apps/public/public.go

```diff
@@ -235,10 +342,23 @@ func WithChallenge(challenge string) AcquireTokenByAuthCodeOption {
 
 // AcquireTokenByAuthCode is a request to acquire a security token from the authority, using an authorization code.
 // The specified redirect URI must be the same URI that was used when the authorization code was requested.
-func (pca Client) AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...AcquireTokenByAuthCodeOption) (AuthResult, error) {
+//
+// Options:
+//   - [WithChallenge]
+//   - [WithTenantID]
+func (pca Client) AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...AcquireByAuthCodeOption) (AuthResult, error) {
 	opts := AcquireTokenByAuthCodeOptions{}
 	for _, o := range options {
-		o(&opts)
+		switch t := o.(type) {
+		case AcquireTokenByAuthCodeOption:
```

Comment:
I looked at this a while and I've come down on the side that we should just make all options here use the new methodology.  Then you can just use the ApplyOptions().  

This makes all options have the same look and feel.  I mean, once you run into one shared option, you gotta deal with it.  Better to make them all consistent.  That also frees us from these switch statements.  

A note as well, you might add in some private doc a link to the explanation on how this all works:
https://blog.devgenius.io/go-call-option-that-can-be-used-with-multiple-methods-6c81734f3dbe